### DISCUSSION TITLE
Conditionnally show tab in panel.tabset in Quarto

### DISCUSSION THREAD

**Discussion by CelieDs** [quarto-dev/quarto-cli#9341] [Q&A]
Hi there!

Thank you for the great tool! I have juste started using it and I was wondering if it was possible to show or hide tabs in the panel.tabset based on specific values present or absent in meta.

Here is my main: 
```
{{< include sections/imports.qmd >}}

{{< include sections/summary.qmd >}}

## Results

### Quality control

::: {.panel-tabset}
## General Results
{{< include general_result.qmd >}}

## Results 1
{{< include result1.qmd >}}

## Results 2
{{< include  result2.qmd >}}`
:::
```

And I would like to do something like this, to conditionnally display the "Results 1" tab or "Results 2" only if the values "result1" or "result2" are present in meta. 

``` 
{{< include sections/imports.qmd >}}

{{< include sections/summary.qmd >}}

## Results

### Quality control

::: {.panel-tabset}
## General Results
{{< include general_result.qmd >}}

::: {.content-hidden unless-meta="result1"}
## Results 1
{{< include result1.qmd >}}
::: 

::: {.content-hidden unless-meta="result2"}
## Results 2
{{< include  result2.qmd >}}`
:::
``` 
This code works when meta doesn't contain "result1" and "result2". However, when meta contains  "result1" or "result2," the sections don't appear in separate tabs as expected but they're displayed in the same tab as "General Results".

Thanks in advance for your help, 

Bests, 

Célie

**Comment by cscheid** [comment]
(@CelieDs FYI, I've converted this into a Q&A discussion on https://github.com/quarto-dev/quarto-cli)

**Comment by mcanouil** [comment]
The header needs to be top-level for tabset to detect it.
Here it is hidden in a div.

The best you can do either as plain markdown or using computation:

```sh
echo 'Here are the results from "result1".' > _results1.qmd
echo 'Here are the results from "result2".' > _results2.qmd
```

````qmd
---
title: "Quarto Playground"
format: html
params:
  results1: true
  results2: false
---

## Markdown

::: {.panel-tabset}
## General Results

Here are the general results.

## Results

:::: {.content-hidden unless-meta="params.results1"}
{{< include _results1.qmd >}}
::::
:::: {.content-hidden unless-meta="params.results2"}
{{< include _results2.qmd >}}
::::
:::


## Computation

::: {.panel-tabset}
## General Results

Here are the general results.

```{r}
#| output: asis
#| echo: false
if (params[["results1"]]) {
  cat("", "## Results 1", "{{< include _results1.qmd >}}", "", sep = "\n")
}
if (params[["results2"]]) {
  cat("", "## Results 2", "{{< include _results2.qmd >}}", "", sep = "\n")
}
```

:::

````

**Comment by sbalci** [comment]
Similar to the answer above, I do with R. 
See how I do with different language and file formats here: 
https://github.com/patolojiatlasi/patolojiatlasi.github.io/blob/main/adipose.qmd

**Reply by mcanouil** [reply]
Thanks for sharing.

Note that you are using the R inline syntax for your code cells options which is specific to the `knitr` engine and will not work elsewhere.
Quarto uses and recommends the YAML syntax.
